contracts/protocol/libraries/logic/MarketplaceLogic.sol

```diff
@@ -469,6 +537,17 @@ library MarketplaceLogic {
 
         reserve.updateState(reserveCache);
 
+        bool willUpdateRateLater = vars.listingToken == vars.creditToken &&
+            vars.creditAmount != 0;
+        if (!willUpdateRateLater) {
+            reserve.updateInterestRates(
+                reserveCache,
+                vars.listingToken,
+                0,
+                vars.isListingTokenPToken ? 0 : vars.supplyAmount
```

Comment:
updateInterestRates in _flashSupplyFor look like should use “liquidityAdded” not “liquidityTaken”